libaleth-interpreter/VM.cpp

```diff
@@ -1356,22 +1356,19 @@ void VM::interpretCases()
             if (m_message->flags & EVMC_STATIC)
                 throwDisallowedStateChange();
 
-            static_assert(
```

Comment:
For the sake of simplicity I got rid of the trick to first charge `sstoreSetGas`, and possibly `sstoreSetGas - sstoreResetGas` afterwards.

To keep it I think we would need to handle pre-Constantinople as an additional special case. In Constantinople we can't know before actually calling `set_storage` whether the change will require DB write, or will be just a dirty write (unless we decide to provide access to "original storage values" to EVM)

Reply:
Now I'm not sure if this is much simpler because it turns out we still need to handle Constantinople here differently (difference in the cost of X->X)
Here's the version with Constantinople as a separate case and keeping the old optimization: https://github.com/ethereum/aleth/pull/5240

Reply:
I think to be safe we should charge more/less the SLOAD cost, so `sstoreUnchangedGas` is good candidate and handles both pre and post Constantinople cases. 
We want to make sure the `sstoreUnchangedGas` is the smallest from all SSTORE possible costs. Would be nice to use `std::min` here but it started be a `constexpr` function in C++14.